homework/index.js

```diff
@@ -30,13 +30,75 @@
     return elem;
   }
 
+  function pageConstructor(data) {
+    data.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
```

Comment:
this line has been used in multiple places so what about using it in the top level of your file?
```javascript
  const root = document.getElementById('root');
  function pageConstructor(data) {
    // Code...
  }
  function main(url) {
    // Code...
  }
```

and in all functions will be the usage of `root` available, also new once will be created will grant that access according the top level scope. ;)